prototype/p2p/network_lifecycle.go

```diff
@@ -0,0 +1,61 @@
+package p2p
+
+import (
+	"pocket/p2p/types"
+	"strings"
+)
+
+func (g *P2PModule) configure(protocol, address, external string, peers []string) {
+	g.protocol = protocol
+	g.address = address
+	g.externaladdr = external
+	g.peerlist = types.NewPeerlist()
+
+	// this is a hack to get going no more no less
```

Comment:
Added a `TODO(derrandz)` for hacks so they are easy to identify, follow and track back. 

Also, for hacks, explain what the hack is doing.